### PR TITLE
Add warning about using `file` with URL paths [ci skip]

### PR DESCRIPTION
**Pull request by mribeirodantas**
Related to https://github.com/nextflow-io/nextflow/issues/2769

### CODE REVIEW

File: docs/script.rst [hunk 1]
```diff
@@ -503,6 +503,23 @@ checkIfExists   When ``true`` throws an exception of the specified path do not e
...
+  assert "$ref" == '/some-bucket/foo.txt'
+
+To have the file including the schema, `toUriString()` should be used instead::
+  assert ref.toUriString() == 's3://some-bucket/foo.txt'
+
+Also, instead of composing paths through string interpolation, the `.resolve` method should be used instead::
+  def dir = file('s3://bucket/some/data/path')
+  def sample = "$dir/sample.bam" // <-- don't do this
+  def sample1 = dir.resolve('sample.bam')
+  def sample2 = dir / 'sample.bam') // <-- the operator `/` can be used to compose paths
```
Comment (pditommaso) [inline docs/script.rst:521]:
```suggestion
  def sample2 = dir / 'sample.bam' // <-- the operator `/` can be used to compose paths
```

**Comment by pditommaso** [comment]
This does not format correctly. Please try in your computer before pushing the changes

**Comment by mribeirodantas** [comment]
I'm sorry for my mistake, Paolo, and thanks for spotting it. It's fixed now.

**Comment by pditommaso** [comment]
Merge manually 2fe23227

**Pull request closed without merging**